### ISSUE TITLE
Map widget missing layers when displayed more than once

### ISSUE THREAD

**Issue by giswqs** [eoda-dev/py-maplibregl#82]
When rendering the map object multiple times in a Jupyter notebook, layers only show up the first time when it is displayed. 

```python
import ipywidgets as widgets

from maplibre import Layer, LayerType
from maplibre.sources import GeoJSONSource
from maplibre.controls import ScaleControl, Marker
from maplibre.ipywidget import MapWidget as Map

# Create a source
earthquakes = GeoJSONSource(
    data="https://docs.mapbox.com/mapbox-gl-js/assets/earthquakes.geojson"
)

# Create a layer
layer_id = "earthquakes"

earthquake_circles = Layer(
    type=LayerType.CIRCLE,
    id=layer_id,
    source=earthquakes,
    paint={"circle-color": "yellow"}
)

# Render map
m = Map(height="200px")
m.add_control(ScaleControl(), position="bottom-left")
m.add_layer(earthquake_circles)
m.add_tooltip(layer_id, "mag")
m.add_marker(Marker(lng_lat=(100.507, 13.745)))
m
```
![image](https://github.com/eodaGmbH/py-maplibregl/assets/5016453/f4fa37ab-98d0-4035-9652-32f5ea70603a)

**Comment by crazycapivara** [comment]
Yep, that's the expected behavior. After the map is rendered, the message queue is cleared!

**Comment by giswqs** [comment]
Is it possible to retain the message queue? 

One use case is that users may want to see the map first before saving it to an HTML file. However, once the map is rendered, saving the map to an HTML file does work properly anymore because the layers are gone.

**Comment by crazycapivara** [comment]
Yes, this seems to be a special use case for the ipywidget. We can store it This also makes it possible to get information about layers etc from the python object even after the map has rendered. So I will keep this issue open.

**Comment by diehl** [comment]
Hi @crazycapivara - I wanted to check in on this issue as it looks like it impacts the Marimo notebook's ability to rerender the visualization when previewing the notebook in application view mode.

**Comment by crazycapivara** [comment]
@diehl I need to check if it is enough to keep the message queue in the case of marimo notebooks, I also need to reset the state. Until it is fixed I encourage you to use the Shiny bindings together with the Shiny VSCode plugin, which also smooth live coding :-)

**Comment by diehl** [comment]
Roger that @crazycapivara :-) Getting this dialed in for Marimo notebooks will be a gamechanger in terms of interactive map development and publishing.

**Comment by crazycapivara** [comment]
@diehl I just played around a little and tt rerenders correct if you deactivate the message queue with

```python
Map._use_message_queue = False
```
(Sorry, this is somehow a hidden feature)

Please see this example:

[ma_notebook_test_app_view_with_widget.py](https://github.com/eoda-dev/py-maplibregl/blob/feature/keep-message-queue/examples/ma_notebook_test_app_view_with_widget.py)

But there is still some strange behavior in marimo notebooks. When switching to app view components added after the widget was created or shown will not be rerendered by marimo, so calling the widget to show up should be the last command.

In one of the next version it is planned to make the map chainable which might help.

**Comment by crazycapivara** [comment]
@diehl Just added a notebook with your example [here](https://github.com/eoda-dev/py-maplibregl/blob/feature/keep-message-queue/_experimental/ma_notebook_test_app_view_with_widget.py).

**Comment by diehl** [comment]
Thanks @crazycapivara! @giswqs Does Stefan's example above provide helpful guidance for making an update in Leafmap?

**Comment by giswqs** [comment]
Sorry for the delay. I have added the `use_message_queue` to the `Map` class in leafmap. The Map objects can now be rendered multiple times without losing layers. Thank you for sharing this hidden feature.
https://github.com/opengeos/leafmap/pull/1078

![image](https://github.com/user-attachments/assets/87508553-f1ee-4e9b-9cbb-124aab777b2a)

**Comment by giswqs** [comment]
Setting `MapWidget._use_message_queue = False` can render the Map objects multiple times, but the exported HTML is missing the data layers.

Setting `MapWidget._use_message_queue = True` can export the HTML properly if the Map is not shown. If the map the shown before exporting it as an HTML, then the HTML will miss the data layers.

**Comment by crazycapivara** [comment]
> Setting `MapWidget._use_message_queue = False` can render the Map objects multiple times, but the exported HTML is missing the data layers.
> 
> Setting `MapWidget._use_message_queue = True` can export the HTML properly if the Map is not shown. If the map the shown before exporting it as an HTML, then the HTML will miss the data layers.

Yes, at the moment this is still the expected behavior as mentioned above. But it will be changed (fixed) in the upcoming release (#129).
